waspc/src/Wasp/Generator/SdkGenerator/EnvValidation.hs

```diff
@@ -43,17 +48,25 @@ genServerEnv spec = return $ C.mkTmplFdWithData tmplPath tmplData
           "defaultServerPort" .= Server.defaultServerPort,
           "enabledAuthProviders" .= (AuthProviders.getEnabledAuthProvidersJson <$> maybeAuth),
           "isEmailSenderEnabled" .= isJust maybeEmailSender,
-          "enabledEmailSenders" .= (EmailSenders.getEnabledEmailProvidersJson <$> maybeEmailSender)
+          "enabledEmailSenders" .= (EmailSenders.getEnabledEmailProvidersJson <$> maybeEmailSender),
+          "envValidationSchema" .= GJI.jsImportToImportJson (extImportToJsImport <$> maybeEnvValidationSchema)
         ]
     maybeAuth = AS.App.auth app
     maybeEmailSender = AS.App.emailSender app
+    maybeEnvValidationSchema = AS.App.server app >>= AS.App.Server.envValidationSchema
     app = snd $ getApp spec
 
-genClientEnv :: Generator FileDraft
-genClientEnv = return $ C.mkTmplFdWithData tmplPath tmplData
+genClientEnv :: AppSpec -> Generator FileDraft
+genClientEnv spec = return $ C.mkTmplFdWithData tmplPath tmplData
   where
     tmplPath = [relfile|client/env.ts|]
-    tmplData = object ["defaultServerUrl" .= Server.defaultDevServerUrl]
+    tmplData =
+      object
+        [ "defaultServerUrl" .= Server.defaultDevServerUrl,
+          "envValidationSchema" .= GJI.jsImportToImportJson (extImportToJsImport <$> maybeEnvValidationSchema)
+        ]
+    maybeEnvValidationSchema = AS.App.client app >>= AS.App.Client.envValidationSchema
```

Comment:
I'd maybe call this `maybeJsImport` and put more logic from line 66 here.

Same applies to the server function